src/fluidplayer.js

```diff
@@ -1911,8 +1911,12 @@ const fluidPlayerClass = function () {
         let initiateVolumebarTimerId = setInterval(initiateVolumebar, 100);
         self.destructors.push(() => clearInterval(initiateVolumebarTimerId));
 
-        if (self.displayOptions.layoutControls.doubleclickFullscreen && !(self.isTouchDevice() || self.displayOptions.layoutControls.controlForwardBackward.doubleTapMobile)) {
-            self.domRef.player.addEventListener('dblclick', self.fullscreenToggle);
+        if (self.displayOptions.layoutControls.doubleclickFullscreen) {
+            if (!self.isTouchDevice()) {
```

Comment:
Looking good. Functionality wise it's working. Now you can simplify this code a bit. If you have the following structure
```
if() { <----------------------------------------------------------------|
    if() { you can always put the logic of this into here with an && ---|

    }
}
```
So if we simplify these `if`'s and `if else`, we have the following
```
if (
    self.displayOptions.layoutControls.doubleclickFullscreen &&
    (!self.isTouchDevice() || !self.displayOptions.layoutControls.controlForwardBackward.doubleTapMobile)
)
```